README.md

```diff
@@ -57,95 +57,388 @@ Some of the skills/concepts this assignment aims to test (and build upon):
+```json
+{
+  "error": false,
+  "errorMsg": null
+}
+```
+
+`403` for Non Existent post or user not logged in
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Post Id"
+}  
+```
+
+
+> PUT `/post/comment`
+
+**Request**
+
+```json
+{
+  "postId": "integer",
+  "text": "string"
+}
+```
+
+**Response**
+`200` On successful comment
+```json
+{
+  "error": false,
+  "errorMsg": ""
+}
+```
+
+`403` for user is not logged in or invalid post id
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Post Id"
+}  
+```
+
+> POST `/post/new`
+
+**Request**
+```json
+{
+  "description_text": "string",
+  "src": "string"
+}
+```
+
+_TODO: Decide if we are doing a file upload or just base64 encode the image_
+
+**Response**
+`200` On successful Post
+```json
+{
+  "error": false,
+  "errorMsg": ""
+}
+```
+
+`403` for user is not logged in or image/description invalid
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Description | Invalid Image"
+}  
+```
+
+
+> GET `/user/feed?p=17`
+
+**Request**
+
+```
+p : a number specifying where to begin fetching posts,
+    the server will response with 10 posts from the feed in the range
+    [10*p,10*p+10). If not specified server will respond with ALL posts
+```
+
+**Response**
+`200` On successful fetch
+```json
+[
+  {
+      "id": 1,
+      "meta": {
+          "author": "Nina",
+          "description_text": "My lounge",
+          "published": "Sat Aug 04 2018 20:19:12 GMT+1000 (Australian Eastern Standard Time)",
+          "likes": []
+      },
+      "thumbnail": "man-small.jpg",
+      "src": "man.jpg"
+  },
+  "..."
+]
+```
+
+`403` for invalid position token or non logged in user
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Position Number"
+}  
+```
+
+> PUT `/user/follow?username=barry`
+
+**Request**
+
+```
+username: username of the user the logged in user wishes to follow
+```
+
+**Response**
+`200` On successful Follow
+```json
+{
+  "error": false,
+  "errorMsg": null
+}
+```
 
-200 OK for successful signup.
-403 for user already exists
+`403` for invalid username or user is not logged in
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Username"
+}  
+```
+
+
+> PUT `/user/unfollow?username=barry`
+
+**Request**
+
+```
+username: username of the user the logged in user wishes to unfollow
+```
+
+**Response**
+`200` On successful UnFollow
+```json
+{
+  "error": false,
+  "errorMsg": null
+}
 ```
 
+`403` for invalid username or user is not logged in
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In | Invalid Username"
+}  
+```
+
+> GET `/user/info`
+
+**Response**
+`200` On successful fetch
+```json
+{
+  "username": "baz",
+  "name": "Barry",
+  "id"  : 1,
+  "posts": [1]
+}
+```
+
+`403` for user is not logged in
+```json
+{
+  "error": true,
+  "errorMsg": "You Are Not Logged In"
+}  
+```
+
+> PUT `/user/update`
```

Comment:
PUT implies update, remove /update

Just: 
`PUT /user` is enough in this instance as only current user can update current user.